Feature/external lb

larsks opened the pull request:
This pull request contains the changes necessary to move from the sharded ingress model to using an external load balancer to expose the cluster externally.

~~Marked as a draft because I am still working on it.~~

There are several commits in this PR; I've tried to structure them logically so that you can review one commit at a time.

larsks committed:
Remove external ingress work

We're rebuilding the cluster to rely on an external loadbalancer,
rather than our sharded ingress solution. This commit removes the
manifests that implemented the external ingress service and the related
network configuration.

Part of ocp-on-nerc/operations#94

larsks committed:
Update apiserver hostname

With the rebuild the cluster hostnames will be in the shift.nerc.mghpcc.org
domain.

Part of ocp-on-nerc/operations#94

larsks committed:
Add Lets Encrypt certificate issuer(s)

Add components to generate namespace-scoped certificate issuers. This
includes both the letsencrypt production and staging servers, so that we
can test certificate generation w/o impacting our rate limits.

larsks committed:
Remove static certificates

We were previously using static TLS certificates for the ingress and api
servers, but now that we have a public hostname and address we can make use
of Lets Encrypt for certificate generation.

larsks committed:
Get route53 credentials from vault

The certificate issuers are configured to use DNS-01 authentication, so we
need appropriate credentials to update entries in route53.

larsks committed:
Add certificate resources for API and ingress servers

larsks committed:
Add issuers to openshift-{config,ingress} namespaces

This allows us to create Certificate resources in the openshift-ingress and
openshift-config namespaces.

jtriley reviewed (commented):
Should we also remove this and update the kustomization.yaml given that we're not handling public ingress on the cluster anymore?

- cluster-scope/overlays/nerc-ocp-prod/nodenetworkconfigurationpolicies/vlan-2180-external.yaml

larsks committed:
Remove metallb configuration

Because we're moving public access to an external load balancer, we no
longer need metallb on nerc-ocp-prod.

larsks committed:
Remove patches to default ingress controller

Since we no longer have a secondary ingress controller for external
access, we don't need to patch the default ingress controller.

larsks commented on cluster-scope/overlays/nerc-ocp-prod/certificates/default-ingress-certificate.yaml:
```diff
@@ -0,0 +1,14 @@
+  namespace: openshift-ingress
+spec:
+  issuerRef:
+    name: letsencrypt-staging-dns01
```
Note that I have intentionally left this configured with the staging server for now; once cert generation has been verified to work correctly we'll swap these to the production issuers.

larsks commented on cluster-scope/overlays/nerc-ocp-prod/certificates/default-api-certificate.yaml:
```diff
@@ -0,0 +1,14 @@
+  namespace: openshift-config
+spec:
+  issuerRef:
+    name: letsencrypt-staging-dns01
```
(See comment below re: staging vs production issuers)

jtriley commented on cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml:
```diff
@@ -6,56 +6,47 @@ commonLabels:
-  path: patches/skipdryrunonmissingresource.yaml
+- path: patches/skipdryrunonmissingresource.yaml
+  target:
+    group: metallb.io
```
Likely harmless but if we're removing metallb do we need this patch?

larsks replied:
It is harmless, it is not necessary...but I guess I'm inclined to leave it here in case we re-introduce metallb at some point.
